@here/harp-examples/src/datasource_geojson_visibility.ts

```diff
@@ -0,0 +1,122 @@
+        const mapView = new MapView({
+            canvas,
+            theme: "resources/berlin_tilezen_base.json",
+            target: NY,
+            zoomLevel: 15.1
+        });
+
+        CopyrightElementHandler.install("copyrightNotice", mapView);
+
+        const mapControls = new MapControls(mapView);
+        mapControls.maxTiltAngle = 50;
+
+        const ui = new MapControlsUI(mapControls, { zoomLevel: "input" });
+        canvas.parentElement!.appendChild(ui.domElement);
+
+        mapView.resize(window.innerWidth, window.innerHeight);
+
+        window.addEventListener("resize", () => {
+            mapView.resize(window.innerWidth, window.innerHeight);
+        });
+
+        const omvDataSource = new VectorTileDataSource({
+            baseUrl: "https://vector.hereapi.com/v2/vectortiles/base/mc",
+            authenticationCode: apikey
+        });
+
+        mapView.addDataSource(omvDataSource);
+
+        omvDataSource.dataSourceOrder = 0;
+
+        class GeoJsonProvider extends DataProvider {
+            readonly feature: turf.helpers.Feature<turf.helpers.Polygon, turf.helpers.Properties>;
+            readonly featureWorldBox = new Box3();
+
+            constructor() {
+                super();
+                this.feature = turf.circle([NY.longitude, NY.latitude], 1000, {
+                    units: "meters"
+                });
+
+                const bbox = turf.bbox(this.feature);
+
+                webMercatorProjection.projectBox(
+                    new GeoBox(
+                        new GeoCoordinates(bbox[1], bbox[0]),
+                        new GeoCoordinates(bbox[3], bbox[2])
+                    ),
+                    this.featureWorldBox
+                );
+            }
+
+            ready(): boolean {
+                return true;
+            }
+
+            async getTile(tileKey: TileKey, _abortSignal?: AbortSignal) {
+                const worldBox = new Box3();
+                webMercatorTilingScheme.getWorldBox(tileKey, worldBox);
+                if (!worldBox.intersectsBox(this.featureWorldBox)) {
+                    return {};
+                }
+                return turf.featureCollection([this.feature]);
+            }
+
+            protected async connect() {}
+            protected dispose(): void {}
+        }
+
+        const featuresDataSource = new VectorTileDataSource({
+            styleSetName: "geojson",
+            dataProvider: new GeoJsonProvider(),
+            addGroundPlane: false
+        });
+
+        featuresDataSource.dataSourceOrder = 1;
+
+        await mapView.addDataSource(featuresDataSource);
+
+        await featuresDataSource.setTheme({
+            styles: [
+                {
+                    styleSet: "geojson",
+                    technique: "fill",
+                    color: "rgba(200,0,0,0.5)",
+                    minZoomLevel: 15,
+                    maxZoomLevel: 16
+                }
```

Comment:
would be good to show here how it works with markers as well.

Reply:
Good point, I will update the example.

Reply:
I modified the example to show a bunch of generated points in the middle of the circle.